src/sentry/migrations/0020_hide_environment_none_20191126.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+def hide_environent_none(apps, schema_editor):
+    """
+    Hide environments that are named none, since they're blacklisted and no longer can be created.
+    """
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+    EnvironmentProject.objects.filter(environment__name='none').update(is_hidden=True)
```

Comment:
This is not a good query. We need to do this iteratively. There are too many EnvironmentProject rows and this will lock the whole table.

I would probably rewrite this to iterate over all Environment rows, filter out in Python the ones that are named "none", then find the corresponding EnvironmentProject rows for that environment id, and mark them hidden.

Runtime is slower, but that's usually better for the database without locking things.